util-app/src/main/scala/com/twitter/app/App.scala

```diff
@@ -105,7 +105,16 @@ trait App extends Closable with CloseAwaitably {
   /**
    * Close `closable` when shutdown is requested. Closables are closed in parallel.
    */
-  final def closeOnExit(closable: Closable): Unit = exits.add(closable)
+  final def closeOnExit(closable: Closable): Unit = {
+    exits.add(closable)
```

Comment:
this may cause the closable to closed twice–this is *probably* ok but it's not part of the closable contract that it is, so let's close it exactly once if possible?  again, mutexes seem to be the right tool here.